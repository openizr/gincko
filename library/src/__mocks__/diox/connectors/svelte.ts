/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { readable } from 'svelte/store/index';

/**
 * diox/connectors/svelte mock.
 */
const connect = vi.fn(() => {
  const state = {
    steps: [{ id: 'start' }, { id: 'end' }],
    loading: process.env.LOADING === 'true',
  };
  return vi.fn(() => readable(state, (set) => {
    set(state);
  }));
});

export default connect;
