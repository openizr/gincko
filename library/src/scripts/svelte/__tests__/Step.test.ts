/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @vitest-environment jsdom
 */

import { render } from '@testing-library/svelte';
import Step from 'scripts/svelte/Step.svelte';

vi.mock('biuty/svelte');
vi.mock('scripts/core/Engine');
vi.mock('scripts/svelte/Field.svelte');

describe('svelte/Step', () => {
  const customComponents = {};
  const userInputs = { test: 'ok' };
  const variables = { var: 'value' };
  const i18n = vi.fn(() => 'TRANSLATED LABEL');
  const onUserAction = vi.fn() as unknown as OnUserAction;

  beforeEach(() => {
    vi.clearAllMocks();
  });

  test('renders correctly - active step', async () => {
    const { container } = render(Step, {
      props: {
        index: 0,
        isActive: true,
        step: {
          id: 'step',
          status: 'success',
          fields: [{
            id: 'Message',
            status: 'initial',
            component: 'Message',
            componentProps: {},
          }],
        },
        i18n,
        variables,
        userInputs,
        onUserAction,
        customComponents,
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - inactive step', async () => {
    const { container } = render(Step, {
      props: {
        index: 0,
        step: {
          id: 'step',
          status: 'success',
          fields: [{
            id: 'Message',
            status: 'initial',
            component: 'Message',
            componentProps: {},
          }, null],
        },
        i18n,
        variables,
        userInputs,
        onUserAction,
        isActive: false,
        customComponents,
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
