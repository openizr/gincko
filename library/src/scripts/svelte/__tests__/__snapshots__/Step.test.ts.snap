// Vitest Snapshot v1

exports[`svelte/Step > renders correctly - active step 1`] = `
<div>
  <div
    class="gincko__step success step active"
    id="step__0"
  >
    <div
      class="gincko__step__fields"
    >
      
      <div
        id="field"
      >
        {"field":{"id":"Message","status":"initial","component":"Message","componentProps":{}},"path":"step.0.Message","isActive":true,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{}}
      </div>
      <!--&lt;Field&gt;-->
      
    </div>
  </div>
  <!--&lt;Step&gt;-->
</div>
`;

exports[`svelte/Step > renders correctly - inactive step 1`] = `
<div>
  <div
    class="gincko__step success step "
    id="step__0"
  >
    <div
      class="gincko__step__fields"
    >
      
      <div
        id="field"
      >
        {"field":{"id":"Message","status":"initial","component":"Message","componentProps":{}},"path":"step.0.Message","isActive":false,"variables":{"var":"value"},"userInputs":{"test":"ok"},"customComponents":{}}
      </div>
      <!--&lt;Field&gt;-->
      
      
      
    </div>
  </div>
  <!--&lt;Step&gt;-->
</div>
`;
